src/Microsoft.Crank.Agent/MstatDumper.cs

```diff
@@ -0,0 +1,201 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Mono.Cecil;
+using Mono.Cecil.Rocks;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+
+namespace Microsoft.Crank.Agent
+{
+    internal static class MstatDumper
+    {
+        internal static DumperResults GetInfo(string path)
+        {
+            var mstats = Directory.EnumerateFiles(path, "*.mstat");
+
+            if (mstats == null || !mstats.Any())
+            {
+                return new DumperResults();
+            }
+
+            using var asm = AssemblyDefinition.ReadAssembly(mstats.First());
+            var globalType = (TypeDefinition)asm.MainModule.LookupToken(0x02000001);
```

Comment:
@MichalStrehovsky @agocke - did we have a library that was able to parse and read mstat information? Can we use that here?

Reply:
We don't have one. https://github.com/dotnet/runtimelab/tree/feature/aot-tools is something that could turn into it at some point.